### PR TITLE
Migrate project from JavaScript to TypeScript

### PR DESCRIPTION
**Pull request by daelynj**
Please let me know if I could have done this PR differently.

Every file has been migrated to TypeScript (.tsx).
Necessary dependencies have been added or updated where necessary.

I have very little TypeScript knowledge, but I believe now is the time to
make the transition, so please help out if you see something; I will learn
as I go.

I should have put commit 6027287 (Update TypingBox Component test)
into its own PR, that was an accident I think.

### CODE REVIEW

File: src/component/gamestats/__tests__/__snapshots__/GameStats.test.tsx.snap [hunk 1]
```diff
@@ -15,7 +15,7 @@ exports[`GameStats renders the expected game stats 1`] = `
     mistakes: 3
   </div>
   <div>
-    accuracy: 92.7%
+    accuracy: 92.6829268292683%
```
Comment (swcraig) [inline src/component/gamestats/__tests__/__snapshots__/GameStats.test.tsx.snap]: Typescript adds this precision?

File: src/component/text/Word.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  word: String;
+  input: string;
+  onCurrentWord: boolean;
+  currentWordSubstring: string;
+  onPastWords: boolean;
+}
+
+export const Word = (props: Props) => {
+  const generateLetters = () =>
+    //for (let letterIndex = 0; letterIndex < props.word.length; letterIndex++)
```
Comment (swcraig) [inline src/component/text/Word.tsx]: If this isn't used we should remove it. Generally we don't want commented out code in our project. We can always add it back if we need it.

File: src/component/__tests__/Board.test.tsx [hunk 3]
```diff
@@ -24,6 +25,8 @@ describe("Board", () => {
     endGame,
     finishTime: 0,
     setFinishTime,
+    checkLetter,
+    incorrectLetters: 0,
```
Comment (swcraig) [inline src/component/__tests__/Board.test.tsx]: I know these were added erroneously, but you are right, these should be in their own commit.

File: src/component/gamestats/__tests__/__snapshots__/GameStats.test.tsx.snap (hunk 1 above)
Comment (daelynj) [reply src/component/gamestats/__tests__/__snapshots__/GameStats.test.tsx.snap]: @swcraig fixed - this was a mistake, rounding was removed. [here](https://github.com/daelynj/typinggame/pull/23/commits/332cd9e6287c3987b0b7f6dfff1278cd1fa6d8e9)

File: src/component/text/Word.tsx (hunk 2 above)
Comment (daelynj) [reply src/component/text/Word.tsx]: @swcraig fixed with a rebase, thank you

File: src/component/__tests__/Board.test.tsx (hunk 3 above)
Comment (daelynj) [reply src/component/__tests__/Board.test.tsx]: @swcraig yeah and something I did with this caused a merge conflict with master so I'll have to deal with the repercussions of sneaking changes in :)
